libs/config/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# We require at least CMake V3.3.2
+cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
+
+project(HPX.config CXX)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+option(HPX_CONFIG_WITH_TESTS "Include tests for config" On)
+
+message(STATUS "config: Configuring")
+
+add_subdirectory(examples)
+add_subdirectory(src)
+add_subdirectory(tests)
+
+install(
+  DIRECTORY include/hpx
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hpx
+  COMPONENT config)
+
+message(STATUS "config: Configuring done")
```

Comment:
We should collect all header files for this module such that those are being added to the generated VS solution files. This also means we have to properly create a source-group for those.

The main problem with source groups in cmake is, that the `source_group` has to be created in the same (cmake) directory as the corresponding `add_library`/`add_executable` is located (see also #3810).